### DISCUSSION TITLE
Copy from Left AND Right?

### DISCUSSION THREAD

**Discussion by ruffin--** [WinMerge/winmerge#1070] [Q&A]
Following the request in [this StackOverflow question](https://stackoverflow.com/q/1126752/1028230):

> Using the TFS merge tool, I can click on the left side and the right side to keep both changes/conflicts. This is needed when I add a method and another team member adds a method in the same place and I need to keep them both. In WinMerge (file merge, not folder), it seems that I can only overwrite the right with the left. **Is it possible to make WinMerge copy the left to the right, and keep the right?** [emph mine]

After reading [this from the help docs](https://manual.winmerge.org/en/Intro_diffs.html#id596814):

> <div  ><div><div><h3  >2.2. Choosing a merge result file<a  name="id596808"></a><a  name="id596804"></a></h3></div></div></div><p >You control where your merge result is preserved. The WinMerge File Compare window does not have a separate result pane, as some merge tools do. So the simplest strategy is to save your changes to one, two or three source files. Press<span> </span><span  ><strong>Ctrl</strong></span>+<span  ><strong>S</strong></span><span> </span>at any time to save the current changes to all files; alternatively, click<span> </span><span  >File</span><span> </span>→<span> </span><span  >Save Left</span>,<span> </span><span  >File</span><span> </span>→<span> </span><span  >Save Middle</span><span> </span>or<span> </span><span  >File</span><span> </span>→<span> </span><span  >Save Right</span><span> </span>to save only one file.</p>

... I tried opening three files with the middle a copy of the left, hoping perhaps I could more easily merge in the middle as a workaround, but it still only allowed taking either the right or left option.

Is there a way to select both with the standard alt-arrow workflow, aka short of "type a new line in file A; now refresh and copy the other line over"?

-------------

For instance:

Original compare state (two one-lined files):
| line number | left | right | line number |
|-|-|-|-|
| 1 | Foo | Bar | 1 |

End state after "Merge left and right":
| line number | left | right | line number |
|-|-|-|-|
| 1 | Foo | Foo | 1 |
| 2 | Bar | Bar | 2 |

-----------------

If not, is that worth a feature request? It would be reasonably easy to implement, I hope; just one extra menu item and shortcut and limited text manipulation logic. If I missed how, AIA!

**Comment by sdottaka** [comment]
Unfortunately, the current WinMerge does not have such a feature.

I sometimes want such a feature, but since the current context menu has many menu items, I would like to add it so that the context menu does not become too complicated.

Note that there is already a Feature Request(https://github.com/WinMerge/winmerge/issues/142) for a similar feature.
